src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -215,25 +215,27 @@ class Peer(
         }
         launch {
             finalWallet.walletStateFlow
-                .distinctUntilChangedBy { it.balance }
-                .collect { wallet ->
-                    logger.info { "${wallet.balance} available on final wallet" }
+                .distinctUntilChangedBy {
+                    Pair(it.balance(includingUnconfirmed = true), it.unconfirmedBalance())
+                }.collect { wallet ->
+                    logger.info { "${wallet.balance(includingUnconfirmed = true)} available on final wallet (${wallet.unconfirmedBalance()} unconfirmed)" }
```

Comment:
With the change above:
```suggestion
                .distinctUntilChangedBy { Pair(it.unconfirmedBalance, it.confirmedBalance) }
                .collect { wallet ->
                    logger.info { "${wallet.totalBalance} available on final wallet (${wallet.unconfirmedBalance} unconfirmed)" }
                }
```